### PR TITLE
Fix removing out of view cells

### LINKED ISSUE #13151: Windowed (Virtual) Notebook DOM structure
## Description

Opening an issue as a follow-up to https://github.com/jupyterlab/jupyterlab/pull/12554.

Manipulating the cells of a new notebook by creating new cells, copying them and deleting them seems to be leaving many hidden nodes in the DOM.

## Reproduce

This can be reproduced with JupyterLab 4.0.0a29 using this gist: https://gist.github.com/jtpio/4e9a3d435d7c3950dc4150c10abbb1d7

https://user-images.githubusercontent.com/591645/193070259-35130d59-c5c5-4719-8530-5c0e7743866a.mp4

## Expected behavior

These nodes with a negative data index might look like an issue.

## Context

This is in the context of updating Notebook 7 to the latest JupyterLab 4.0 pre-release: https://github.com/jupyter/notebook/pull/6539

Prior to https://github.com/jupyterlab/jupyterlab/pull/12554, the notebook DOM structure was simple enough so it could be customized with a bunch of CSS: https://github.com/jupyter/notebook/blob/73646334817ffaf3dd6adde7a746a2cc2ff1f9cf/packages/notebook-extension/style/base.css#L27-L92

The new structure / logic of the virtual notebook implementation will require some updates to be done in Notebook 7.

cc @fcollonval

### PR DESCRIPTION
**Pull request by fcollonval**
## References

Fixes #13151

## Code changes

For windowed notebook, we need to delete the output area remaining in the DOM for code cells out of the view port.

## User-facing changes

N/A

## Backwards-incompatible changes

N/A

### CODE REVIEW

File: galata/test/jupyterlab/windowed-notebook.test.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  await Promise.all([
+    firstCell.waitForElementState('visible'),
+    page.mouse.wheel(0, -1200)
+  ]);
+
+  // Check that the input area is back
+  expect(await firstCell.waitForSelector('.jp-InputArea')).toBeDefined();
+});
+
+test('should detached a markdown code cell when scrolling out of the viewport', async ({
```
Comment (jtpio) [inline galata/test/jupyterlab/windowed-notebook.test.ts:56]:
```suggestion
test('should detach a markdown code cell when scrolling out of the viewport', async ({
```

File: galata/test/jupyterlab/windowed-notebook.test.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  let found = true;
+  try {
+    await h!.waitForSelector(mdCellSelector, { timeout: 150 });
+  } catch (r) {
+    found = false;
+  }
+  expect(found).toEqual(false);
+});
+
+test('should reattached a markdown code cell when scrolling back into the viewport', async ({
```
Comment (jtpio) [inline galata/test/jupyterlab/windowed-notebook.test.ts:79]:
```suggestion
test('should reattach a markdown code cell when scrolling back into the viewport', async ({
```

File: docs/source/developer/performance.rst [hunk 3]
```diff
@@ -4,6 +4,26 @@
 Performance tricks
 ==================
 
+Windowed list
+-------------
+
+A list windowing algorithm is coded in a dedicated widget ``WindowedList`` provided by ``@jupyterlab/ui-components``.
```
Comment (jtpio) [inline docs/source/developer/performance.rst:10]:
```suggestion
A list windowing algorithm is implemented in a dedicated widget ``WindowedList`` provided by ``@jupyterlab/ui-components``.
```

**Review by jtpio — APPROVED** [review]
Thanks!

Just left a couple of minor comments.

**Comment by fcollonval** [comment]
Thanks for the review @jtpio

**Pull request merged**